libs/ui/popover/src/lib/trigger/popover-trigger.directive.ts

```diff
@@ -200,15 +210,36 @@ export class TsPopoverTriggerDirective implements OnInit, OnDestroy, OnChanges,
     this.popover.referenceObject = this.viewContainerRef.element.nativeElement;
     this.setContentProperties(this.popover);
 
-    // eslint-disable-next-line deprecation/deprecation
-    (fromEvent<MouseEvent>(this.elementRef.nativeElement, 'click'))
-      .pipe(
-        untilComponentDestroyed(this),
-      ).subscribe(event => {
-        this.toggle();
-        event.stopPropagation();
-        event.preventDefault();
-      });
+    switch (this.showTrigger) {
```

Comment:
Seems like it might be better to have the setter for the `showTrigger` input wire this up?

Reply:
Hmm... what'd be the main difference for it being here or in setter? My thought was that the event wiring up should be happening when component starts up.

Reply:
I'm on board with that 👍